File: BusinessLogic/Models/Persistency/CardBase.cs
using System;

namespace kRetro.BusinessLogic.Models.Persistency
{
    public enum CardType{
        Good,
        Bad,
        Unknown
    }
    public class CardBase
    {
        public CardType Type {get; private set;}

        public CardBase(){
            Type = CardType.Unknown;
        }        
        public CardBase(CardType type){
            Type = type;
        }
        public int Id {get;set;}
        public string Message {get;set;}
        public User User {get;set;}
        public DateTime CreationDateTime {get;set;}
        public bool Visible {get;set;}   
        public int? ClusterId {get;set;}            
    }
}

File: ClientApp/models/persistency/cardBase.ts
import {User} from './user';

export enum CardType{
    Good,
    Bad
}

export class CardBase {
    private _type: CardType
    public get type(): CardType {return this._type};

    public constructor(type: CardType ){
        this._type = type;
    }

    public id: number | undefined;
    public message: string | undefined;
    public user: User | undefined;
    public creationDateTime: Date | undefined;
    public visible: boolean | undefined;
}